File: cloudsystem-base/src/main/java/de/hydracloud/base/command/defaults/HelpCommand.java
package de.hydracloud.base.command.defaults;

import de.hydracloud.base.Base;
import de.hydracloud.base.command.CloudCommand;

@CloudCommand.Command(name = "help", description = "All commands and help description")
public final class HelpCommand extends CloudCommand {

    @Override
    public void execute(Base base, String[] args) {
        final var manager = Base.getInstance().getCommandManager();

        base.getLogger().log("§7All possible commands(§b" + manager.getCachedCloudCommands().size() + "§7):");
        manager.getCachedCloudCommands().values().stream().distinct().forEach(it -> base.getLogger()
            .log("§b" + it.getName() + getAliases(it) + " §7- " + it.getDescription()));
    }

    private String getAliases(CloudCommand command) {
        return command.getAliases().length == 0 ? "" : "§7(§b" + String.join(", ", command.getAliases()) + "§7)";
    }

}

File: src/hydracloud/cloud/command/impl/HelpCommand.php
<?php

namespace hydracloud\cloud\command\impl;

use hydracloud\cloud\command\argument\def\StringArgument;
use hydracloud\cloud\command\Command;
use hydracloud\cloud\command\CommandManager;
use hydracloud\cloud\command\sender\ICommandSender;

final class HelpCommand extends Command {

    public function __construct() {
        parent::__construct("help", "List all commands");
        $this->addParameter(new StringArgument(
            "command",
            true
        ));
    }

    public function run(ICommandSender $sender, string $label, array $args): bool {
        $command = $args["command"] ?? null;
        $commands = $command === null ? CommandManager::getInstance()->getAll() : (($tmp = CommandManager::getInstance()->get($command)) === null ? CommandManager::getInstance()->getAll() : [$tmp]);

        foreach ($commands as $command) {
            $sender->info("§b" . $command->getName() . " §8- §r" . $command->getDescription() . " §8- §b" . $command->getUsage());
        }
        return true;
    }
}